### DISCUSSION TITLE
Is it possible to add {MODE=VIRTUAL} to the options under "Perform Auto-Type"

### DISCUSSION THREAD

**Discussion by dani5612** [keepassxreboot/keepassxc#11138] [Q&A] [answered]
Hey everyone! 

I have a set of entries that is mostly used in virtual machines. I know I can set up a custom Auto - Type sequence that allows me to just add {MODE=VIRTUAL}{_mySequence_}. However, since I only can add one custom Auto - Type sequence, I have to choose if I want to make that sequence {MODE=VIRTUAL}{USERNAME} OR {MODE=VIRTUAL}{PASSWORD} or {MODE=VIRTUAL}{USERNAME}{TAB}{PASSWORD}{ENTER}. These options already exist under the Perform Auto - Type menu for all other entries. 

Is it possible to add more options to the Perform Auto - Type menu? or maybe make it so my entry is set up in virtual mode so it automatically adds the {MODE=VIRTUAL} to all the default Auto - Type options?

Thank you for your time :)

**Comment by droidmonkey** [comment]
You can setup as many window association sequences as you want, you are not limited to one.

**Reply by dani5612** [reply]
I am sorry :( Maybe I am doing something wrong.

This is the way I my entries right now:

![image](https://github.com/user-attachments/assets/7ed7ed90-6b00-4bf1-887a-fb6f1fcb2730)

![image](https://github.com/user-attachments/assets/4a15b993-16e8-4b0b-b1de-7523b92a48d8)

And I am mostly using entries like these in VMWare Remote Console windows

![image](https://github.com/user-attachments/assets/4f1fc670-e526-4400-8f60-c843dd445b3b)

What I tried to do was add more Windows Associations that would fit my VMWare Remote Console windows. One with {MODE=VIRTUAL}{PASSWORD}{ENTER} and the other with {MODE=VIRTUAL}{USERNAME}

![image](https://github.com/user-attachments/assets/5964e4de-a4d8-4222-ae5f-e0167635cb67)

Whenever I call Auto Type using Ctrl + Shift + V from KeePassXC, it correctly calls the {MODE=VIRTUAL}{USERNAME}{TAB}{PASSWORD}{ENTER}
![image](https://github.com/user-attachments/assets/c1c7ff2a-2a53-4d35-b189-acff88e6992d)
![image](https://github.com/user-attachments/assets/1e4cd2ee-dca7-406e-8f9a-94ff96f767fb)

but... now I am unsure about how to call the Windows associations from the Auto - Type settings. Am I supposed to change the options back to "Inherit default Auto - Type sequence from the group"?  Am I using the Window Title option wrong with the "*"?

![image](https://github.com/user-attachments/assets/621d7c93-d897-4fe2-9469-4d46232840fd)

If I call the Auto - Type function with the settings shown right above, it calls the default {USERNAME}{TAB}{PASSWORD}{ENTER} without asking me about anything about the windows associations.

I am sorry if I am missing something that is obvious.

**Reply by droidmonkey** [reply, marked as answer]
https://keepassxc.org/docs/KeePassXC_UserGuide#_performing_global_auto_type

**Reply by dani5612** [reply]
Massive face palm. Thank you, I totally missed this.

In case someone stumbles across this same problem in the future:

Make sure you assign a "Global Auto - Type shortcut" in the Application Settings

![image](https://github.com/user-attachments/assets/d76dd00e-c732-425d-acf7-8af0652715da)

If you are trying to use this in a Virtual Machine Remote Console, make sure you click on the window that contains the view to the VM, do not go into the VM or your stokes will be captured by the console.

![image](https://github.com/user-attachments/assets/a3c56c80-0e39-4d35-b37f-dbb31078fa85)

Once the window to your VM is selected, just press the Global Auto - Type shortcut that you created earlier (In my case Ctrl + Shift + PgUp). This will bring any entries that match the title of the window. In this case, the window ends with - VMware Remote Console, so the following entries show up
 
![image](https://github.com/user-attachments/assets/97a975d7-f62b-496b-8219-f73d33dcbfa2)
![image](https://github.com/user-attachments/assets/b3d8263a-8de2-4b9a-8cba-360345744361)

I had to hide some extra entries that showed up since I made the window title "*", making them match with any window name.

**Reply by droidmonkey** [reply]
Hehe glad you are all set now
